[Zhu Jingxi] iP

Cheezeblokz opened the pull request.

Cheezeblokz committed:
- Added skeletal greeting with robot name, prompt and exit greeting.

Cheezeblokz committed:
- Added ASCII art for robot name
- Added and personalised robot prompt statements
- Added functionality of echoing user input
- Added exit program when given command 'bye'

Cheezeblokz committed:
- Added comments for code
- Added 'list' command for listing all previous user inputs
- Removed echoing of user input

Cheezeblokz committed:
- Added Task class for encapsulation and object representation of tasks in the task list
- Added extra commands for marking and unmarking tasks in the task list
- Updated displaying of task information to include marked/unmarked condition

bnjm2000 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+                if(tasksIndex == 0) {
+                    System.out.println("No item stored in your list! :o");
+                }
+                else {
```
"else" statement should be right after "}"

bnjm2000 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+        return isMarked;
+    }
+
+    public void setMarked(boolean marked) {
+        isMarked = marked;
```
should be "this.isMarked = true;" since it is a boolean variable

kaijie0102 replied:
I can see that you are  going for a "toggle method" that goes from marked to unmarked and unmarked to marked.
Consider making method name less confusing.

Alternatively have another method called unsetMarked()

bnjm2000 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+                }
+            }
+            // if not unique command, taken as adding a new Task
+            else {
```
avoid [arrowhead style code](https://blog.codinghorror.com/flattening-arrow-code/)

bnjm2000 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+        // introduction and initial prompt for user input
+        System.out.print("Hello! My name is:\n" + name + "\n\n" +
+                "What can I do for you today? :)\n" + ">>");
+
```
awesome customisation

kaijie0102 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+            }
+            // if 'mark' or 'unmark' command is given strictly with an integer, if integer is valid
+            // mark or unmark the corresponding task in tasks
+            else if(userInput.toLowerCase().matches("(mark|unmark) \\d+")) {
```
Consider using a switch-case block to improve readability.
switch(userInput){
case "list":
case "bye":
.
.
.
}

kaijie0102 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
-}
+
+    // private function to print a stream of underscores for partitioning robot conversation
+    private static void partition() {
+        System.out.println("____________________________________________________________");
+    }
```
consider changing a trivial function to a constant

PARTITION = "_______________________"

J-Y-Yan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+        partition();
+
+        // name of the robot in ASCII art
+        String name = " _____ _                       _     _       _         \n" +
+                "/  __ \\ |                     | |   | |     | |        \n" +
+                "| /  \\/ |__   ___  ___ _______| |__ | | ___ | | __ ____\n" +
+                "| |   | '_ \\ / _ \\/ _ \\_  / _ \\ '_ \\| |/ _ \\| |/ /|_  /\n" +
+                "| \\__/\\ | | |  __/  __// /  __/ |_) | | (_) |   <  / / \n" +
+                " \\____/_| |_|\\___|\\___/___\\___|_.__/|_|\\___/|_|\\_\\/___|";
+
+        // introduction and initial prompt for user input
+        System.out.print("Hello! My name is:\n" + name + "\n\n" +
+                "What can I do for you today? :)\n" + ">>");
```
I like your welcome message. Very interesting.

J-Y-Yan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+        Scanner scanner = new Scanner(System.in);
+
+        // string to monitor current user input
+        String userInput = scanner.nextLine();
```
I guess using the name "input" will be concise.

J-Y-Yan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+                }
+                // not in range of number of tasks
+                else {
+                    System.out.println("Invalid integer input! :(");
```
Maybe you could improve the error message, such as "there is no task at the kth position".

J-Y-Yan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    //setters and getters
+    public boolean isMarked() {
```
Using "getMark", "getStatus" could be more precise as a getter function.

J-Y-Yan reviewed (commented):
LGTM. The indentation is precise. Maybe the naming of variable or method can be improved. The code overall is very readable with lots of comments. I like his customized setting of robots.

madhanse commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+            // if not unique command, taken as adding a new Task
+            else {
+                tasks[tasksIndex] = new Task(userInput); // Store user input into array
+                tasksIndex++; // Increase String array index
```
Minimize the usage of comments and remove the comments that are very obvious. For example, increment of tasksIndex already means that you are incrementing this index.

Cheezeblokz committed:
Added dummy classes for Task children classes (Deadline, Todo, Event)

Cheezeblokz committed:
Improved user input processing code to flatten arrow codes

Cheezeblokz committed:
Added user command for adding Todo, Event and Deadline as Tasks
Added error message and gatekeeping for invalid commands

Cheezeblokz committed:
Code quality improvment, addition of steps for easier code readability

Cheezeblokz committed:
Added DukeException class to handle Duke specific errors
Added error handling for missing information when adding Todo, Event or Deadline

Cheezeblokz committed:
Merge branch 'branch-Level-5'

# Conflicts:
#	src/main/java/Duke.java

Cheezeblokz committed:
Implemented packaging for Duke class and secondary classes

Cheezeblokz committed:
Modification of data structure for Task storage from list to ArrayList

Cheezeblokz committed:
Added delete command to remove tasks from the list

Cheezeblokz committed:
Added file IO capabilities for saving data locally on the machine

Cheezeblokz committed:
Merge branch 'branch-Level-6'

Cheezeblokz committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java

Cheezeblokz committed:
Edited IO file directory

Cheezeblokz committed:
Added classes for handling different functionalities for Duke
Refactored relevant code from Duke class to the respective classes

Cheezeblokz committed:
Adopted OOP for TaskList and streamlined codes for OOP implementation

Cheezeblokz committed:
Adopted OOP for FileStorage and Duke class and streamlined codes for OOP implementation

Cheezeblokz committed:
Adopted OOP for UIHandler class and streamlined codes for OOP implementation

Cheezeblokz committed:
Added find command to look for tasks in the list with the same name as the given string

Cheezeblokz committed:
Merge pull request #1 from Cheezeblokz/branch-Level-9

Added find command to look for tasks in the list with the same name a…

Cheezeblokz committed:
Added JavaDoc comments for classes FileStorage, Parser and TaskList

Cheezeblokz committed:
Merge pull request #2 from Cheezeblokz/branch-A-JavaDoc

Added JavaDoc comments for classes FileStorage, Parser and TaskList

Cheezeblokz committed:
Added User Guide.

Cheezeblokz committed:
Updated User Guide.

Cheezeblokz committed:
Updated User Guide links.
